### DISCUSSION TITLE
git graph and jupyter not work

### DISCUSSION THREAD

**Discussion by vipwp** [gitpod-io/openvscode-server#157] [Q&A] [answered]
Does this issue occur when all extensions are disabled?: Yes/No

- VS Code Version: 1.61.0
- OS Version: CentOS Linux release 7.6.1810 x86_64

Steps to Reproduce:

1. install jupyter v2021.9.100 and git graph v1.30.0
2. jupyter and git graph not work expectedly on code server but works well on vscode client.

**Comment by akosyakov** [comment]
I could not reproduce the issue:
<img width="1035" alt="Screenshot 2021-10-13 at 09 11 09" src="https://user-images.githubusercontent.com/3082655/137084644-3ef525da-ede2-4ce8-9490-a4c482376d46.png">
<img width="1026" alt="Screenshot 2021-10-13 at 09 11 55" src="https://user-images.githubusercontent.com/3082655/137084637-42cf8000-f425-4836-b85e-8516d76797dc.png">

Are you sure that you installed all required tooling in your image for these extensions to work? What do you mean by they don't work?

**Comment by bhavitsharma** [comment]
I have the same openvscode/vscode version and git graph, jupyter notebooks stopped working. Maybe the extensions updated and broke because I remember these extensions working before. I will take a look again.

**Comment by vipwp** [comment]
@akosyakov  Thanks for the reply.

The required tooling is ok. 
The git graph view is blank. The jupyter can not add code.  There are same errors in the web console when open them.
![image](https://user-images.githubusercontent.com/1547414/137426645-3dd4e6e3-0152-40cc-ab9f-f2bbb81b2bc0.png)

Web Browser: chrome@91.0
Local OS:windows 10 x64
Remote OS: centos 7.2
Remote Architecture: x64
Version: 1.62.0

**Comment by akosyakov** [comment, marked as answer]
Webviews requires service workers which browser allow only in secure context, i.e. https or localhost. Do you run on http but not localhost? I don't expect it to work in such configuration without disabling security in the browser.

**Reply by bhavitsharma** [reply]
thanks, that explains it! btw, any high level idea on how one can run the server.sh via https? (maybe ngnix?)
Also, it'll be great if you can add some info about extensions not working properly with http in the README.

**Reply by vipwp** [reply]
> Webviews requires service workers which browser allow only in secure context, i.e. https or localhost. Do you run on http but not localhost? I don't expect it to work in such configuration without disabling security in the browser.

Thanks for the reply.

**Reply by akosyakov** [reply]
> thanks, that explains it! btw, any high level idea on how one can run the server.sh via https? (maybe ngnix?)
Also, it'll be great if you can add some info about extensions not working properly with http in the README.

Could you file an issue to provide guidelines with https behind proxy? I would recommend caddy, it provides better defaults to handle web socket connections for instance. But it is rather matter of taste.

**Reply by bhavitsharma** [reply]
thanks, I will file an issue. I can even contribute once I have setup my own! I'll also look into caddy.

**Reply by jeanp413** [reply]
You can also checkout this discussion https://github.com/gitpod-io/openvscode-server/discussions/96#discussioncomment-1410639 which contains an apache2 proxy configuration

**Comment by Marandil** [comment]
Sorry for unearthing this thread, but I'm running into a similar problem with Jupyter and - in fact other WebViews - failing to load at all, in localhost context.

The error is the following:
```
Error loading webview: Error: Could not register service workers: NotSupportedError: Failed to register a ServiceWorker for scope ('https://1ac9oufdd263r1f0834ts467jrtjt36iln457g13tc78d8jg652k.vscode-cdn.net/insider/ef65ac1ba57f57f2a3961bfe94aa20481caca4c6/out/vs/workbench/contrib/webview/browser/pre/') with script ('https://1ac9oufdd263r1f0834ts467jrtjt36iln457g13tc78d8jg652k.vscode-cdn.net/insider/ef65ac1ba57f57f2a3961bfe94aa20481caca4c6/out/vs/workbench/contrib/webview/browser/pre/service-worker.js?v=4&vscode-resource-base-authority=vscode-resource.vscode-cdn.net&remoteAuthority=localhost:8081'): The user denied permission to use Service Worker..
```

I'm guessing the cause is somehow related to vscode pulling scripts from CDN which is not in a "secure/trusted" context, is that right? (I'm OOL with regards to browser security features).

C.f. 
![image](https://user-images.githubusercontent.com/3922660/196705255-e7db8ae2-646e-4fd1-8862-c8310feff878.png)

Browser: Chrome 106.0.5249.119 (64bit)
Client OS: Windows 10 Pro 21H2
Host OS: Ubuntu 20.04.3 (running directly, without Docker)
openvscode-server version: v1.72.2-linux-x64

**Reply by jeanp413** [reply]
Maybe you have “Block third-party cookies” setting enabled in your browser settings?

**Reply by Marandil** [reply]
Yeah, that seems to be the case. Changing the setting to "incognito only" seems to have helped. Is there any way to fix this without sacrificing privacy for *all the other websites*? 😅 (I've elected to block third party cookies consciously.)
Something like a self-hosted mode, or a proxy?

**EDIT** and besides just adding "localhost" as an exemption from the rule; although that seems to be fine as a stopgap solution.

**Reply by jeanp413** [reply]
You can allow specific websites in `Sites that can always use cookies`

**Reply by Marandil** [reply]
Yeah, I found that (see Edit), but I was wondering if a solution that doesn't require adding this exception by future users/clients (ofc then from a https context and not localhost) is possible / exists.

TYVM for helping me solve this though!

**Reply by gihrig** [reply]
"Maybe you have “Block third-party cookies” setting enabled in your browser settings?"
In my case the problem was Privacy Badger. Potentially any privacy extension or add blocker could be a problem.
